addons/stock/models/stock_picking.py

```diff
@@ -267,6 +267,13 @@ def get_action_picking_type_operations(self):
     def get_stock_picking_action_picking_type(self):
         return self._get_action('stock.stock_picking_action_picking_type')
 
+    @api.constrains('sequence_id')
+    def _check_sequence_code(self):
+        for picking_type in self:
+            sequence_name = picking_type.sequence_id.name
```

Comment:
To avoid making multiple search. You can replace them by a `read_group` with group by (company, name) and check the `count` of each groups

Reply:
@Whenrow I changed it as you request